core/arch/opnd_shared.c

```diff
@@ -1986,7 +1986,7 @@ reg_is_extended(reg_id_t reg)
             (reg >= REG_START_x64_8 && reg <= REG_STOP_x64_8) ||
             (reg >= DR_REG_START_XMM + 8 && reg <= DR_REG_STOP_XMM) ||
             (reg >= DR_REG_START_YMM + 8 && reg <= DR_REG_STOP_YMM) ||
-            (reg >= DR_REG_START_ZMM + 8 && reg <= DR_REG_STOP_ZMM) ||
+            (reg >= DR_REG_START_ZMM + 16 && reg <= DR_REG_STOP_ZMM) ||
```

Comment:
Should be +8, please revert.

Reply:
@hgreving2304 whether **reg_is_extended()** does not checks `reg` is high register or not? For zmm registers high registers are zmm16-zmm31.

Reply:
reg_is_extended is used to determine < 8 registers in dimode. It's the same for AVX-512 (32-bit still has 8 registers). We may need a new function like reg_is_super_extended (bad name) for AVX-512 (but less architecture specific).